internal/trafficlogs/logger.go

```diff
@@ -0,0 +1,161 @@
+/*
+ * @license
+ * Copyright 2023 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package trafficlogs
+
+import (
+	"fmt"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/log"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/log/field"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/timeutils"
+	"github.com/google/uuid"
+	"github.com/spf13/afero"
+	"net/http"
+	"net/http/httputil"
+	"os"
+	"path"
+	"sync"
+)
+
+type FileBasedLogger struct {
+	fs               afero.Fs
+	requestFilePath  string
+	responseFilePath string
+	requestLogFile   afero.File
+	responseLogFile  afero.File
+	lock             sync.Mutex
+}
+
+func NewFileBased() *FileBasedLogger {
+	tl := &FileBasedLogger{
+		fs:               afero.NewOsFs(),
+		requestFilePath:  path.Join(log.LogDirectory, timeutils.TimeAnchor().Format("20060102-150405")+"-"+"req.log"),
+		responseFilePath: path.Join(log.LogDirectory, timeutils.TimeAnchor().Format("20060102-150405")+"-"+"resp.log"),
+	}
+
+	return tl
+}
+
+func (l *FileBasedLogger) Log(req *http.Request, reqBody string, resp *http.Response, respBody string) error {
+	l.lock.Lock()
+	defer l.lock.Unlock()
+
+	requestId := ""
+	if l.isRequestLogEnabled() {
```

Comment:
these will currently always be true won't they? Nothing is exported, so the `NewFileBased` constructor is the only way to ever get a logger

Reply:
yes, this is just a logical leftover of the previous code will remove that check